### PR TITLE
Feature/thread safe

### PR DESCRIPTION
**Pull request by ChaoKuan-Lin**
Support run with background thread.

### CODE REVIEW

**Review by schwannden — APPROVED** [review]

File: src/decanter/core/context.py [hunk 1]
```diff
@@ -9,6 +9,14 @@
 
 logger = logging.getLogger(__name__)
 
+def get_or_create_eventloop():
```
Comment (yuwaj6) [inline src/decanter/core/context.py:12]: Looks like this will loop forever?
I tried the change and this will fail when tried to create CoreClient....

**Review by yuwaj6 — CHANGES_REQUESTED** [review]
I tried the change and this will fail when tried to create CoreClient....

File: src/decanter/core/context.py (hunk 1 above)
Comment (schwannden) [reply src/decanter/core/context.py:12]: oh yeah, it does looks like an infinite loop XD

**Review by schwannden — APPROVED** [review]

**Review by yuwaj6 — APPROVED** [review]

**Comment by ChaoKuan-Lin** [comment]
Support background thread run with SDK

**Pull request merged**